lstore/table.py

```diff
@@ -252,5 +260,33 @@ def __merge(self):
         for i in range(self.total_num_columns):
             self.tail_pages_prev_merge[i] = len(self.tail_pages[i]) - 1
         
+    def serialize(self):
+        """Returns table metadata as a JSON-compatible dictionary"""
+        return {
+            "num_columns": self.num_columns,
+            "key_index": self.key,
+            "page_directory": self.page_directory,
+            "rid_index": self.rid_index
+        }
+    
+    def preload_pages(self):
```

Comment:
No need for db to load pages into bufferpool. Algorithm would be more complex if db has to figure out what pages to load into bufferpool